src/lib.rs

```diff
@@ -100,6 +100,7 @@ impl URLBuilder {
             for (param, value) in self.params.iter() {
                 url_params.push_str(format!("{}={}&", param, value).as_str());
             }
+            url_params.pop();
```

Comment:
Let's leave a comment here explaining why we need to always `pop` at the end—it's not obvious at first glance

Reply:
That's understandable. I've just added a comment.